cws-service/src/main/java/jpl/cws/console/WorkerMonitorBackgroundThread.java

```diff
@@ -74,7 +80,50 @@ public void run() {
 					//
 					workersThatWentDown = schedulerDbService.detectDeadWorkers(THRESHOLD_MILLIS_FOR_DEAD_WORKER);
 				}
-				
+
+
+
+				//log.debug("DOWNTIME LOG INFO: " + numDaysAfterRemoveDeadWorkers);
+				//log.debug("DOWNTIME LOG INFO: " + historyDaysToLive);
+				//String numsTest = "5";
+				//int THRESHOLD_MILLIS_FOR_ABANDONED_WORKER = Integer.parseInt(numsTest) * 86400000;
+				//log.warn("DOWNTIME TIME: -- " + THRESHOLD_MILLIS_FOR_ABANDONED_WORKER);
+
+
+				// ---------------------------------
+				// CHECK FOR DOWN WORKERS & DELETE WORKERS THAT ARE PAST THE ABANDONED WORKER LIMIT "num_days_after_to_remove_abandoned_workers"
+				// ---------------------------------
+				int THRESHOLD_MILLIS_FOR_ABANDONED_WORKER = Integer.parseInt(numDaysAfterRemoveDeadWorkers) * 86400000;
+
+				List<Map<String,Object>> workersThatAreAbandoned = schedulerDbService.detectAbandonedWorkers(THRESHOLD_MILLIS_FOR_ABANDONED_WORKER);
+
+				//log.debug("DOWNTIME LOG INFO: " + numDaysAfterRemoveDeadWorkers);
+				//String val1 = schedulerDbService.showJodaTime(THRESHOLD_MILLIS_FOR_ABANDONED_WORKER);
+				//log.debug("DOWNTIME TIME: -- " + val1);
+
+				while (!workersThatAreAbandoned.isEmpty()) {
+					//
+					// Get first worker in List
+					//
+					Map<String,Object> worker = workersThatWentDown.get(0);
+					String workerId = worker.get("id").toString();
+
+					// Check lastHeartbeatTime against "num_days_after_to_remove_abandoned_workers" value
+
+					schedulerDbService.deleteAbandonedWorkers(workerId);
+
+					log.warn("Detected (and removed abandoned row in DB) cws_worker '" + workerId +
+						"' (threshold milliseconds since last worker heartbeat is " +
+						THRESHOLD_MILLIS_FOR_ABANDONED_WORKER);
```

Comment:
make this human readable 
"worker's last heartbeat was: <timestamp>" or similar